spec/unit/providers/vsphere_spec.rb

```diff
@@ -159,6 +159,30 @@
         expect(result).to eq(expected_vm_list)
       end
     end
+
+    context 'given a pool folder with a folder and vms' do
+      let(:expected_vm_list) {[
+        { 'name' => 'vm1'},
+        { 'name' => 'vm2'},
+        { 'name' => 'vm3'}
+      ]}
+      let(:folder_object2) { mock_RbVmomi_VIM_Folder({ :name => 'pool2'}) }
+      before(:each) do
+        expected_vm_list.each do |vm_hash|
+          mock_vm = mock_RbVmomi_VIM_VirtualMachine({ :name => vm_hash['name'] })
+          # Add the mocked VM to the folder
+          folder_object.childEntity << mock_vm
+        end
+        folder_object.childEntity << folder_object2
+        expect(subject).to receive(:find_folder).with(pool_config['folder'],connection,datacenter_name).and_return(folder_object)
+      end
+
+      it 'should return the vms without the folder' do
+        result = subject.vms_in_pool(poolname)
```

Comment:
that looks good, do you want it to also return VMs inside the folder 'pool2' or is it supposed to skip these?

Reply:
No, it's supposed to skip those and just evaluate the objects that are VMs in the base folder object. Previously it would treat the folder as though it was a VM and try to discover and destroy it.

Reply:
Yes I've seen that before. I wonder why we would have subfolders though